index.js

```diff
@@ -220,7 +220,7 @@ function requestAsEventEmitter(opts) {
 				ee.emit('error', new got.RequestError(err, opts));
 			});
 
-			ee.on('request', req => {
+			ee.once('request', req => {
```

Comment:
This is the offending line. The other ones are just safeguarding for future memory leaks.

Reply:
Yep, all good.